File: src/main/java/pt/joaomneto/titancompanion/adventure/impl/SOTSAdventure.java
package pt.joaomneto.titancompanion.adventure.impl;

import java.io.BufferedWriter;
import java.io.IOException;

import pt.joaomneto.titancompanion.R;
import pt.joaomneto.titancompanion.adventurecreation.impl.fragments.sots.SOTSMartialArt;

import android.os.Bundle;
import android.view.Menu;

public class SOTSAdventure extends TFODAdventure {

	int currentHonour = -1;
	int willowLeafArrows = -1;
	int bowelRakerArrows = -1;
	int armourPiercerArrows = -1;
	int hummingBulbArrows = -1;
	SOTSMartialArt skill = null;

	static Integer FRAGMENT_EQUIPMENT = 2;
	static Integer FRAGMENT_NOTES = 3;
	
	public SOTSAdventure() {
		super();
		fragmentConfiguration.clear();
		fragmentConfiguration.put(FRAGMENT_VITAL_STATS, new AdventureFragmentRunner(R.string.vitalStats, "pt.joaomneto.titancompanion.adventure.impl.fragments.sots.SOTSAdventureVitalStatsFragment"));
		fragmentConfiguration.put(FRAGMENT_COMBAT, new AdventureFragmentRunner(R.string.fights, "pt.joaomneto.titancompanion.adventure.impl.fragments.sots.SOTSAdventureCombatFragment"));
		fragmentConfiguration.put(FRAGMENT_EQUIPMENT, new AdventureFragmentRunner(R.string.goldEquipment, "pt.joaomneto.titancompanion.adventure.impl.fragments.sots.SOTSAdventureEquipmentFragment"));
		fragmentConfiguration.put(FRAGMENT_NOTES, new AdventureFragmentRunner(R.string.notes, "pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureNotesFragment"));
	}

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		try {
			super.onCreate(savedInstanceState);

		} catch (Exception e) {
			e.printStackTrace();
		}

	}

	@Override
	public boolean onCreateOptionsMenu(Menu menu) {
		getMenuInflater().inflate(R.menu.adventure, menu);
		return true;
	}

	@Override
	public void storeAdventureSpecificValuesInFile(BufferedWriter bw) throws IOException {
		bw.write("honour=" + getCurrentHonour() + "\n");
		bw.write("skill=" + getSkill().name() + "\n");
		if (SOTSMartialArt.KYUJUTSU == skill) {
			bw.write("armourPiercerArrows=" + getArmourPiercerArrows() + "\n");
			bw.write("bowelRakerArrows=" + getBowelRakerArrows() + "\n");
			bw.write("hummingBulbArrows=" + getHummingBulbArrows() + "\n");
			bw.write("willowLeafArrows=" + getWillowLeafArrows() + "\n");
		}
	}

	@Override
	protected void loadAdventureSpecificValuesFromFile() {
		setCurrentHonour(Integer.valueOf(getSavedGame().getProperty("honour")));
		setSkill(SOTSMartialArt.valueOf(getSavedGame().getProperty("skill")));
		if (SOTSMartialArt.KYUJUTSU == skill) {
			setArmourPiercerArrows(Integer.parseInt(getSavedGame().getProperty("armourPiercerArrows")));
			setBowelRakerArrows(Integer.parseInt(getSavedGame().getProperty("bowelRakerArrows")));
			setHummingBulbArrows(Integer.parseInt(getSavedGame().getProperty("hummingBulbArrows")));
			setWillowLeafArrows(Integer.parseInt(getSavedGame().getProperty("willowLeafArrows")));
		}
	}

	public int getCurrentHonour() {
		return currentHonour;
	}

	public void setCurrentHonour(int currentHonour) {
		this.currentHonour = currentHonour;
	}

	public int getWillowLeafArrows() {
		return willowLeafArrows;
	}

	public void setWillowLeafArrows(int willowLeafArrows) {
		this.willowLeafArrows = willowLeafArrows;
	}

	public int getBowelRakerArrows() {
		return bowelRakerArrows;
	}

	public void setBowelRakerArrows(int bowelRakerArrows) {
		this.bowelRakerArrows = bowelRakerArrows;
	}

	public int getArmourPiercerArrows() {
		return armourPiercerArrows;
	}

	public void setArmourPiercerArrows(int armourPiercerArrows) {
		this.armourPiercerArrows = armourPiercerArrows;
	}

	public int getHummingBulbArrows() {
		return hummingBulbArrows;
	}

	public void setHummingBulbArrows(int hummingBulbArrows) {
		this.hummingBulbArrows = hummingBulbArrows;
	}

	public SOTSMartialArt getSkill() {
		return skill;
	}

	public void setSkill(SOTSMartialArt skill) {
		this.skill = skill;
	}

}

File: src/main/java/pt/joaomneto/titancompanion/adventure/impl/SOTSAdventure.kt
package pt.joaomneto.titancompanion.adventure.impl

import android.os.Bundle
import android.view.Menu
import pt.joaomneto.titancompanion.R
import pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureNotesFragment
import pt.joaomneto.titancompanion.adventure.impl.fragments.sots.SOTSAdventureCombatFragment
import pt.joaomneto.titancompanion.adventure.impl.fragments.sots.SOTSAdventureEquipmentFragment
import pt.joaomneto.titancompanion.adventure.impl.fragments.sots.SOTSAdventureVitalStatsFragment
import pt.joaomneto.titancompanion.adventurecreation.impl.fragments.sots.SOTSMartialArt
import pt.joaomneto.titancompanion.util.AdventureFragmentRunner
import java.io.BufferedWriter
import java.io.IOException

class SOTSAdventure : TFODAdventure(
    arrayOf(
        AdventureFragmentRunner(R.string.vitalStats, SOTSAdventureVitalStatsFragment::class),
        AdventureFragmentRunner(R.string.fights, SOTSAdventureCombatFragment::class),
        AdventureFragmentRunner(R.string.goldEquipment, SOTSAdventureEquipmentFragment::class),
        AdventureFragmentRunner(R.string.notes, AdventureNotesFragment::class)
    )
) {

    var currentHonour = -1
    var willowLeafArrows = -1
    var bowelRakerArrows = -1
    var armourPiercerArrows = -1
    var hummingBulbArrows = -1
    var skill: SOTSMartialArt? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        try {
            super.onCreate(savedInstanceState)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.adventure, menu)
        return true
    }

    @Throws(IOException::class)
    override fun storeAdventureSpecificValuesInFile(bw: BufferedWriter) {
        bw.write("honour=" + currentHonour + "\n")
        bw.write("skill=" + skill!!.name + "\n")
        if (SOTSMartialArt.KYUJUTSU == skill) {
            bw.write("armourPiercerArrows=" + armourPiercerArrows + "\n")
            bw.write("bowelRakerArrows=" + bowelRakerArrows + "\n")
            bw.write("hummingBulbArrows=" + hummingBulbArrows + "\n")
            bw.write("willowLeafArrows=" + willowLeafArrows + "\n")
        }
    }

    override fun loadAdventureSpecificValuesFromFile() {
        currentHonour = Integer.valueOf(savedGame.getProperty("honour"))
        skill = SOTSMartialArt.valueOf(savedGame.getProperty("skill"))
        if (SOTSMartialArt.KYUJUTSU == skill) {
            armourPiercerArrows = Integer.parseInt(savedGame.getProperty("armourPiercerArrows"))
            bowelRakerArrows = Integer.parseInt(savedGame.getProperty("bowelRakerArrows"))
            hummingBulbArrows = Integer.parseInt(savedGame.getProperty("hummingBulbArrows"))
            willowLeafArrows = Integer.parseInt(savedGame.getProperty("willowLeafArrows"))
        }
    }

    companion object {

        internal var FRAGMENT_EQUIPMENT: Int? = 2
        internal var FRAGMENT_NOTES: Int? = 3
    }
}
